File: corporate-hotel-booking/src/company/company_service.py
from src.company.employee import Employee
from src.company.employee_repository import EmployeeRepository


class CompanyService:
    def __init__(self, employee_repository: EmployeeRepository):
        self.__employee_repository = employee_repository

    def add_employee(self, company_id: str, employee_id: str) -> None:
        employee = Employee(company_id=company_id, employee_id=employee_id)
        self.__employee_repository.add_employee(employee=employee)

    def delete_employee(self, employee_id: str) -> None:
        self.__employee_repository.delete_employee(employee_id=employee_id)

File: corporate-hotel-booking-java/src/main/java/com/github/ambersariya/CompanyService.java
package com.github.ambersariya;

import com.github.ambersariya.core.CompanyId;
import com.github.ambersariya.core.EmployeeId;
import com.github.ambersariya.employee.Employee;
import com.github.ambersariya.employee.EmployeeRepository;

public class CompanyService {

    private final EmployeeRepository employeeRepository;

    public CompanyService(EmployeeRepository employeeRepository) {
        this.employeeRepository = employeeRepository;
    }

    public void deleteEmployee(EmployeeId employeeId) {
        employeeRepository.deleteEmployee(employeeId);
    }

    public void addEmployee(EmployeeId employeeId, CompanyId companyId) {
        var employee = new Employee(employeeId, companyId);
        employeeRepository.addEmployee(employee);
    }
}
